File: Web_Visualization/src/test/java/com/gatech/project/utils/HBaseUtilsTest.java
package com.gatech.project.utils;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

import com.gatech.project.params.Params;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
class HBaseUtilsTest {
    @Test
    public void HBaseUtilsTest() {
        long value = 2;
        HBaseUtils connect = HBaseUtils.getInstance(Params.QUORUM, Params.PORT);
        connect.put(Params.TABLENAME, Params.ROWKEY_TEST, Params.COLUMN_FAMILY, Params.COLUMN, value);

        Map<String, Long> map = connect.query(Params.COLUMN_FAMILY, Params.COLUMN, Params.TABLENAME, Params.ROWKEY_TEST);
        for (Map.Entry<String, Long> entry : map.entrySet()) {
            System.out.println(entry.getKey() + " : " + entry.getValue());
            if(entry.getKey() == "20200328") {
                assert(entry.getValue() == value);
            }
        }
    }

}

File: Logs_Analysis/src/test/scala/com/gatech/project/utils/HBaseUtilsTest.scala
package com.gatech.project.utils

import org.junit._
import Assert._
import com.gatech.project.dao.LanguageSearchCountDAO.{cf, qualifer}
import com.gatech.project.params.Params
import org.apache.hadoop.hbase.client.Get
import org.apache.hadoop.hbase.util.Bytes

@Test
class HBaseUtilsTest {

  @Test
  def HBaseUtilsTest() = {
    val value = 3
    HBaseUtils.setConf(Params.quorum, Params.port)
    if(!HBaseUtils.existTable(Params.table)) {
      print("table not existed yet")
      HBaseUtils.createTable(Params.table, Params.cf)
    }
    print("table already existed yet")
    HBaseUtils.singlePut(Params.table, Params.rowkey, Params.cf, Params.column, Params.value)
    val setVal = HBaseUtils.singleGet(Params.table, Params.rowkey, Params.cf, Params.column)
    assert(value == setVal)
  }
}
